src/game/Game.java

```diff
@@ -107,98 +103,164 @@ public int getPlayerTurn() {
 		return this.currentBoard.pieceAt(coor);
 	}
 
 	/**
 	 * 
-	 * @param row
-	 * @param column
-	 * @param dir
-	 *            0: up, 1: right, 2: down, 3: left
-	 * @return returns true if the move made successfully, otherwise returns false
+	 * @param moveToMake
+	 * @return
 	 */
-	public boolean move(int row, int column, int dir) {
-		MoveCommand moveToMake;
-		if (!isValidMoveFromSquare(row, column))
-			return false;
-		switch (dir) {
-		case 0:
-			// Moving UP
-			moveToMake = new MoveUp(this.currentBoard);
-			return makeMove(moveToMake, row, column);
-		case 1:
-			// Moving RIGHT
-			moveToMake = new MoveRight(this.currentBoard);
-			return makeMove(moveToMake, row, column);
-		case 2:
-			// Moving DOWN
-			moveToMake = new MoveDown(this.currentBoard);
-			return makeMove(moveToMake, row, column);
-		case 3:
-			// Moving LEFT
-			moveToMake = new MoveLeft(this.currentBoard);
-			return makeMove(moveToMake, row, column);
-		default:
+	public boolean move(BoardState boardState, Coordinate start, Coordinate dest) {
+		AbstractPiece piece = this.currentBoard.getPieceAt(start);
+		if (isFrozen(start)) {
 			return false;
 		}
+		if ((piece instanceof Rabbit)) {
+			if (((piece.getOwner() == Owner.values()[0]) && (dest.equals(start.up())))
+					|| ((piece.getOwner() == Owner.values()[1]) && (dest.equals(start.down())))) {
+				// Cannot move a Rabbit backwards unless it has been dragged
+				return false;
+			}
+		}
+		if (start.isOrthogonallyAdjacentTo(dest)
+				&& this.getBoardState().getPieceAt(start).getOwner() == this.getOwner()) {
+			RegularMove moveToMake = new RegularMove(boardState, start, dest);
+			this.currentBoard = moveToMake.execute();
+			this.moves.add(moveToMake);
+			endMove();
+			return true;
+		}
+		return false;
+	}
+
+	private boolean isFrozen(Coordinate pieceToMove) {
+		if (!isNextToFriendlyPiece(pieceToMove, this.getOwner())
+				&& isNextToEnemyPiece(pieceToMove, this.getOtherOwner())) {
+			return true;
+		}
+		return false;
 	}
 
+	private boolean isNextToEnemyPiece(Coordinate pieceToMove, Owner player) {
+		for (int i = -1; i < 2; i++) {
+			for (int j = -1; j < 2; j++) {
+				Coordinate coor = new Coordinate(i + pieceToMove.getX(), j + pieceToMove.getY());
+				if (this.getPieceAt(coor) != null) {
+					if (coor.isValid() && !coor.equals(pieceToMove)
+							&& this.getPieceAt(coor).getOwner() == player
+							&& this.getPieceAt(coor).isStrongerThan(this.getPieceAt(pieceToMove))) {
+						return true;
+					}
+				}
+			}
+		}
+		return false;
+	}
+	
+	private boolean isNextToFriendlyPiece(Coordinate pieceToMove, Owner player) {
+		for (int i = -1; i < 2; i++) {
+			for (int j = -1; j < 2; j++) {
+				Coordinate coor = new Coordinate(i + pieceToMove.getX(), j + pieceToMove.getY());
+				if (this.getPieceAt(coor) != null) {
+					if (coor.isValid() && !coor.equals(pieceToMove)
+							&& this.getPieceAt(coor).getOwner() == player) {
+						return true;
+					}
+				}
+			}
+		}
+		return false;
+	}
+	
```

Comment:
Looks similar, but is a bit difficult to refactor out into a singular method used twice due to the "this.getPieceAt(coor).isStrongerThan(this.getPieceAt(pieceToMove)" piece in isNextToEnemyPiece().